src/thor/costmatrix.cc

```diff
@@ -286,40 +288,31 @@ void CostMatrix::SourceToTarget(Api& request,
                                                                     .edgeid(),
                                                                 tile),
                                 time);
-      auto* pbf_date_time = matrix.mutable_date_times()->Add();
-      *pbf_date_time = dt_info.date_time;
-
-      auto* pbf_time_zone_offset = matrix.mutable_time_zone_offsets()->Add();
-      *pbf_time_zone_offset = dt_info.time_zone_offset;
-
-      auto* pbf_time_zone_name = matrix.mutable_time_zone_names()->Add();
-      *pbf_time_zone_name = dt_info.time_zone_name;
+      *matrix.mutable_date_times(connection_idx) = dt_info.date_time;
+      *matrix.mutable_time_zone_offsets(connection_idx) = dt_info.time_zone_offset;
+      *matrix.mutable_time_zone_names(connection_idx) = dt_info.time_zone_name;
```

Comment:
I also changed this whole logic, it became unbearable now that we're "randomly" processing connections, i.e. not in sequence anymore. I realized that `RepeatedPtrFields` don't have a `Set()` method but also their `operator()` methods aren't const, so we can just assign new values. probably that's quite a bit worse for performance, but I doubt it's very measurable wrt to the whole expansion.

Reply:
basically now we're initializing all `repeated string` fields with `""` in `reserve_pbf_arrays()`, just like the other fields which have `Set()`, in a way reverting some of what was done in #4530